.github/workflows/release.yml

```diff
@@ -120,22 +120,19 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           DRY_RUN: true
 
-      - name: "3. Determine docker tags"
-        run: echo ::set-env name=DOCKER_TAGS::$(echo latest, ${{ steps.bump_dry.outputs.tag }})
+      - name: "3. Login to docker hub"
+        run: "echo '${{ secrets.DOCKER_HUB_TOKEN }}' | docker login --username ${{ secrets.DOCKER_HUB_USER }} --password-stdin"
 
-      - name: "3-1. Determine docker tags for hotfix"
+      - name: "4-1. Publish hotfix docker image"
         if: contains(github.event.head_commit.message, '#hotfix')
-        run: echo ::set-env name=DOCKER_TAGS::$(echo ${{ steps.bump_dry.outputs.tag }})
+        run: |
+          ./gradlew dockerTagsPush -P version=${{ steps.bump_dry.outputs.tag }}
+
+      - name: "4-2. Publish non-hotfix docker image"
+        if: ${{ !contains(github.event.head_commit.message, '#hotfix') }}
```

Comment:
what happens in case of #patch in commit message?

Reply:
actually also - what happens in case of hotfix itself?

Reply:
Maybe versioning logic should be documented somewhere (not for this pr) as it looks like it is not easy to deduce, at least for me.

Reply:
> what happens in case of #patch in commit message?

There was not any specific behaviour for #patch commit before my changes as far as I can see

Reply:
> actually also - what happens in case of hotfix itself?

In hotfix we skip publishing docker image with tag 'latest' since hotfix (according to my conversation with @endrjuskr ) refers to change in one of previous versions

Reply:
This is definitely something we have to concretely specify somewhere. Because the two latest hotfixes changed the latest release..

Reply:
Hotfixe is something that applies to older version that we still support. For instance v1 and v2. In our cases it was more like patch or minor, so that is why we released latest.